lib/ui/painting.dart

```diff
@@ -2296,25 +2296,44 @@ class Gradient extends Shader {
   /// If `matrix4` is provided, the gradient fill will be transformed by the
   /// specified 4x4 matrix relative to the local coordinate system. `matrix4` must
   /// be a column-major matrix packed into a list of 16 values.
+  /// 
+  /// If `focal` is provided and not equal to `center` or `focalRadius` is 
+  /// provided and not equal to 0.0, the generated shader will be a two point
+  /// conical radial gradient, with `focal` being the center of the focal 
+  /// circle and `focalRadius` being the radius of that circle. If `focal` is
+  /// provided and not equal to `center`, at least one of the two offsets must
+  /// not be equal to [Offset.zero].
   Gradient.radial(
     Offset center,
     double radius,
     List<Color> colors, [
     List<double> colorStops,
     TileMode tileMode = TileMode.clamp,
-    Float64List matrix4
+    Float64List matrix4,
+    Offset focal,
+    double focalRadius
```

Comment:
oh man, we should have used named arguments. this has become sad.

Reply:
Yeah :( but that'd be a a breaking change.  I suppose it could be another method and deprecate this one....

Reply:
we could do another method, but as you point out in the commit message, that's got its own problems.

Reply:
let's just do a good job of the framework wrapper, then this particular method won't matter much.